worker/worker_test.go

```diff
@@ -0,0 +1,124 @@
+package worker
+
+import (
+	"context"
+	"time"
+
+	rhpv2 "go.sia.tech/core/rhp/v2"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/blake2b"
+	"lukechampine.com/frand"
+)
+
+type (
+	test interface {
+		Fatal(...any)
+	}
+
+	testWorker struct {
+		t test
```

Comment:
I'd make this a `*TT` and maybe move that type out of `cluster.go` so that we can access it from anywhere. It is quite useful in testing.